src/core.rs

```diff
@@ -322,6 +321,45 @@ pub async fn states_edit(
     Ok(plan)
 }
 
+/// Clear Data Products so then can be rerun
+pub async fn clear_edit(
```

Comment:
@coderabbitai Please write me the following unit test for the following functions:
- For `core.rs`, please write the following unit test for `clear_edit()`:
  - Can I clear a data product to a `queued` state if the data set is unpaused and all its upstream dependencies are satisfied?
  - If I clear a data product and have all its downstream children updates to a `waiting` state?
  - If I try to clear a non-existent data product, do I get an error?
  - If I try to clear “disabled” data product, do I get an error?